plugins/http_plugin/http_plugin.cpp

```diff
@@ -938,12 +951,15 @@ namespace eosio {
          my->thread_pool->stop();
       }
 
+      // release http_plugin_impl_ptr shared_ptrs captured in url handlers
+      my->url_handlers.clear();
```

Comment:
This isn't quite enough.  The thread pool's `io_context` may contain unexecuted handlers that hold `shared_ptrs` to the `http_plugin_impl`.

Reply:
I'm not following. The `io_context` destructor will destroy all the unexecuted handlers that hold `shared_ptrs` to `http_plugin_impl`.
https://www.boost.org/doc/libs/develop/doc/html/boost_asio/reference/io_context/_io_context.html

Do you just mean the comment is not clear? Or do you see an issue here?

Reply:
You're only stopping the thread pool, not resetting it.  This leaves a `shared_ptr` cycle.

Reply:
Ah. Thanks!